tls/s2n_ktls_io.c

```diff
@@ -405,3 +406,35 @@ int s2n_sendfile(struct s2n_connection *conn, int in_fd, off_t offset, size_t co
     *blocked = S2N_NOT_BLOCKED;
     return S2N_SUCCESS;
 }
+
+int s2n_ktls_read_full_record(struct s2n_connection *conn, uint8_t *record_type)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(record_type);
+
+    /* This method copies data into conn->in, so is intended for control messages
```

Comment:
When you say this method is "intended for control messages rather than application data" doesn't that mean that we should optimize for receiving control messages? Like, we're not going to recv application data very often according to this comment.

Reply:
True, but we do have to be able to handle application data. If I set the buffer size to S2N_ALERT_LENGTH (2), reading any application data is going to take a LOT of syscalls :)

Reply:
And to handle post-handshake messages we'd also need to set it higher than S2N_ALERT_LENGTH. How much higher is tricky, since NewSessionTicket messages don't have an easy size, but they're probably <1k.

But I guess by "optimize for control messages" you're suggesting the first option from https://github.com/aws/s2n-tls/pull/4199#discussion_r1324885861, but choose a size that fits post-handshake messages and only resize for application data.